crypto/src/lib.rs

```diff
@@ -113,14 +113,14 @@ impl FromStr for Algorithm {
 /// Options for key generation
 #[cfg(not(feature = "ffi_import"))]
 #[derive(Debug, Clone)]
-pub enum KeyGenOption {
+pub enum KeyGenOption<K = PrivateKey> {
     /// Use random number generator
     #[cfg(feature = "rand")]
     Random,
     /// Use seed
     UseSeed(Vec<u8>),
     /// Derive from private key
-    FromPrivateKey(Box<PrivateKey>),
+    FromPrivateKey(K),
```

Comment:
this seems weird. There is a variant `FromPrivateKey` but you can parametrize it with something that's not a private key?

Reply:
The idea was to parametrize it with the private key of concrete type related to the exact algorithm